### ISSUE TITLE
Store "Driver-Run-Receipts" on Disk

### ISSUE THREAD

**Issue by nlordell** [cowprotocol/services#888]
The protocol is constantly running auctions and sending them to `driver`s that may or may not be chosen as the winner. Over the course of each of these auctions, there is important information that they may want to keep a record of.

This issue capture the work to store in some form of persistent storage, per auction:
- [ ] The input auction model that is sent from the `autopilot` to the `driver` _(useful for debuggability)_
- [ ] The abstract auction model that is sent from the `driver` to the `searcher` _(useful for debuggability)_
- [ ] The abstract settlement model that is sent from the `searcher` to the `driver` _(useful for debuggability)_
- [ ] The signed proposed solution and objective value that is sent from the `driver` to the `autopilot` _(useful for accountability)_
- [ ] The signed request for `/reveal` and `/execute` that is sent from the `autopilot` to the `driver`; this may become very important in case of any disputes about whether or not a `driver` was told to execute some solution by an `autopilot` _(useful for accountability)_

Basically, we want to have a detailed receipt, in persistent storage, of every auction run for both debuggability as well as accountability regarding respecting the protocol rules.

**Comment by vkgnosis** [comment]
> useful for accountability

It sounds like this should happen in Autopilot if we need it for accountability. Debuggability can happen in the driver.

> on disk

I guess you mean any persistent storage instead of literally on disk.

**Comment by nlordell** [comment]
> It sounds like this should happen in Autopilot if we need it for accountability.

Yes, it should - but that is already captured by https://github.com/cowprotocol/services/issues/886. I think that storing this in the driver as well is important for accountability in case: 
- We say to a solver "why did you execute that settlement?! You weren't supposed to!"
- They say to us "you told me I should, here is a signed message".

For this specific case, drivers should keep their own "as-run-receipts" for accountability in case of disputes.

> I guess you mean any persistent storage instead of literally on disk.

Yes.

**Comment by vkgnosis** [comment]
I'm unsure how much sense embedded/local file databases make with our kubernetes setup where containers do not have persistent files. Then we'd have to have this file on some other network service and mount it into the container. We had a setup kind of like this a couple of years which felt awkward. Maybe we just didn't do it right but I feel something over a network connection makes more sense. In our case the natural choice would be Postgres again.

If we don't want to require this for people running the driver themself we could have a switch that appends logs to a file.

**Comment by fleupold** [comment]
- autopilot <> driver auction: tracked in #1752 
- solver_engine <> solver auction: stored in logs, for persistence should be stored by the solvers themselves
- settlement model as well as the other accounting items are stored in logs, whose retention should be sufficient for disputes.
